### ISSUE TITLE
Make `ignore` platform/os agnostic?

### ISSUE THREAD

**Issue by NeekSandhu** [kaelzhang/node-ignore#48]
First thank you for making such a sick library!

https://runkit.com/neeksandhu/runkit-npm-ignore

Snippet above shows how `dist/*` matches `dist/index.js` but won't match `dist\build`. 

A windows path being matched against a unix glob that is.

As we all know, and as mentioned in [one of the previous issues here](https://github.com/kaelzhang/node-ignore/issues/5#issuecomment-169949286), almost all applications and programs just work given above scenario; windows paths <-> unix paths

I think there aren't any side effects if `ignore` is made platform agnostic as far as type of slashes are concerned.

We can very well remove `process.platform === 'win32'` thing in code just by using `[\/\\]+` instead.

I'm trying to use `ignore` in a webapp where I'd love to have this feature :)

**Comment by NeekSandhu** [comment]
And just to add on why this would be awesome, we should think of `ignore` as a **"path"** matching library as opposed to **"string"** matching thing.

The "semantic" meaning instead of "syntactic".

`ignore` should be the answer to the question:
Does this **pattern** match the following **resource**?

And `ignore` saying no to the above question when given `dist/**` and `dist\scripts\index.js` respectively doesn't sound very right.

**Comment by kaelzhang** [comment]
I thought just convert the path `dist\script\index.js` is dangerous, because some linux os supports directory names or file names that contain back slashs. For example, `foo\bar/baz.js` is a valid file path on Mac OS, in which a javascript file `bar.js` is inside the one-depth directory `foo\bar`.

**Comment by NeekSandhu** [comment]
Crap 😧!!

That changes everything. I didn't know about this till now. I have published a bunch of stuff in past that blindly separates paths on `[\\\/]` and moves on with the day. I'll have to fix all those things.

I kept hating `win32` for being the only one bad kid for not following the convention of using forward slashes like others but at least it isn't ambiguous in the type of slashes it uses in path(s).

Well, I guess I'll retract this request unless you have anything in mind.
